lib/MusicBrainz/Server/WebService/XMLSerializer.pm

```diff
@@ -281,19 +283,27 @@ sub _serialize_release_group
             Interview => 5
         );
 
-        my ($fallback) =
+        my ($fallback_name) =
             nsort_by { $fallback_type_order{$_} }
                 grep { exists $fallback_type_order{$_} }
                     map { $_->name }
                         $release_group->all_secondary_types;
+        my ($fallback_gid) =
+            nsort_by { $fallback_type_order{$_} }
+                grep { exists $fallback_type_order{$_} }
```

Comment:
So something like this?

```perl
my $fallback =
    nsort_by { $fallback_type_order{$_} }
        grep { exists $fallback_type_order{$_} }
            $release_group->all_secondary_types;

if ($fallback) {
    $attr{type} = $fallback->name;
    $attr{"type-id"} = $fallback->gid;
} else {
    $attr{type} = $release_group->primary_type->name;
    $attr{"type-id"} = $release_group->primary_type->gid;
}
```

Though I'm having trouble understanding when `$fallback` is actually used.